test/cql-pytest/test_null.py

```diff
@@ -66,19 +66,17 @@ def test_insert_null_key(cql, table1):
 
 def test_primary_key_in_null(cql, table1):
     '''Tests handling of "key_column in ?" where ? is bound to null.'''
-    s = random_string()
-    cql.execute(f"INSERT INTO {table1} (p,c) VALUES ('{s}', '{s}')")
     with pytest.raises(InvalidRequest, match='null value'):
         cql.execute(cql.prepare(f"SELECT p FROM {table1} WHERE p IN ?"), [None])
     with pytest.raises(InvalidRequest, match='null value'):
-        cql.execute(cql.prepare(f"SELECT p FROM {table1} WHERE p='{s}' AND c IN ?"), [None])
+        cql.execute(cql.prepare(f"SELECT p FROM {table1} WHERE p='' AND c IN ?"), [None])
     with pytest.raises(InvalidRequest, match='Invalid null value for IN restriction'):
-        cql.execute(cql.prepare(f"SELECT p FROM {table1} WHERE p='{s}' AND (c) IN ?"), [None])
+        cql.execute(cql.prepare(f"SELECT p FROM {table1} WHERE p='' AND (c) IN ?"), [None])
 
 # Cassandra says "IN predicates on non-primary-key columns (v) is not yet supported".
 def test_regular_column_in_null(scylla_only, cql, table1):
     '''Tests handling of "regular_column in ?" where ? is bound to null.'''
-    s = random_string()
-    cql.execute(f"INSERT INTO {table1} (p,c) VALUES ('{s}', '{s}')")
+    # Without any rows in the table, SELECT will shortcircuit before evaluating the WHERE clause.
```

Comment:
This is interesting. Are you sure about this? How can SELECT know if there are any rows in the table before it executes a query??